### DISCUSSION TITLE
Mask and Detection

### DISCUSSION THREAD

**Discussion by gniknalu** [roflcoopter/viseron#197] [Ideas]
In the README, it says about a Mask:

`Masks are used to exclude certain areas in the image from triggering motion.`

But what about detection? If it excludes areas from **triggering motion**, should it also exclude from **detection**?

As you know I'm using the dev branch - updated as of yesterday - but it will detect objects and record a snapshot and video, when the objects are in my mask area. The workaround - which worked but isn't feasible - is to create a zone (instead of a mask) and have it only record when there is a kangaroo detected (I'm exaggerating) or something that you wouldn't see in that zone. Yet I think a mask is the appropriate place to  have a "not detect nor record" rule. Do you agree?

**Thank you**

**Comment by roflcoopter** [comment]
Yes i do agree with you.
Motion masks should stay the way they are now.
However you should be able to specify a mask under object_detection aswell which filters out any objects in it.

**Reply by roflcoopter** [reply]
I created a feature request issue to track this

**Comment by gniknalu** [comment]
Here is something in line with this issue for reference. Running the dev code (from 2021.06.10)

There is a defined mask here. But it still detects (what it **thinks** is) a person.
![2021-06-10-024804-person_detect-w-mask](https://user-images.githubusercontent.com/3652809/121591588-4bc55880-c9ff-11eb-915d-28f36b77d6fa.jpg)
![2021-06-10-024804-person_detect](https://user-images.githubusercontent.com/3652809/121591599-4ec04900-c9ff-11eb-8681-8fc789dbae90.jpg)

**Comment by gniknalu** [comment]
Rather than a mask, I created a region and a label for "person" and "trigger_recorder: false". This will do for now.

**Comment by roflcoopter** [comment]
I just pushed a new version of `dev` where you now can add a mask under `object_detection` just like you can under `motion_detection`

**Reply by gniknalu** [reply]
> I just pushed a new version of `dev` where you now can add a mask under `object_detection` just like you can under `motion_detection`

I'll give it a try! **Thank you for everything you're doing!**

**Reply by gniknalu** [reply]
> I just pushed a new version of `dev` where you now can add a mask under `object_detection` just like you can under `motion_detection`

**It Works Great!** Thank you!
